WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -456,12 +461,16 @@ public static void openEditorForReblog(
             site.getSiteId()
         );
 
-        Intent editorIntent = new Intent(activity, EditPostActivity.class);
-        editorIntent.putExtra(EditPostActivityConstants.EXTRA_REBLOG_POST_TITLE, post.getTitle());
-        editorIntent.putExtra(EditPostActivityConstants.EXTRA_REBLOG_POST_QUOTE, post.getExcerpt());
-        editorIntent.putExtra(EditPostActivityConstants.EXTRA_REBLOG_POST_IMAGE, post.getFeaturedImage());
-        editorIntent.putExtra(EditPostActivityConstants.EXTRA_REBLOG_POST_CITATION, post.getUrl());
-        editorIntent.setAction(EditPostActivityConstants.ACTION_REBLOG);
+        EditorLauncherParams params = new EditorLauncherParams.Builder(site)
+                .reblogPostTitle(post.getTitle())
+                .reblogPostQuote(post.getExcerpt())
+                .reblogPostImage(post.getFeaturedImage())
+                .reblogPostCitation(post.getUrl())
+                .reblogAction(EditPostActivityConstants.ACTION_REBLOG)
+                .source(reblogSource)
```

Comment:
The original implementation didn't set this here and set it in `addNewPostForResult` instead. This wouldn't have been a problem, but I've removed it in 4198b77427c11c0b9e4d7d3de08d02f5621d2770 to match the previous implementation.